src/StatusQ/Controls/StatusInput.qml

```diff
@@ -419,13 +423,22 @@ Item {
 
         StatusBaseText {
             id: errorMessage
-            visible: !!text && !statusBaseInput.valid
-            height: visible ? contentHeight : 0
+            visible: {
+                if (!text)
+                    return false;
+
+                if ((root.validationMode == StatusInput.ValidationMode.OnlyWhenDirty && statusBaseInput.dirty) ||
+                    root.validationMode == StatusInput.ValidationMode.Always)
+                return !statusBaseInput.valid;
```

Comment:
This needs a tab I think, or brackets, to show it's with the if above.

Reply:
fixed